sdk/cosmos/azure-cosmos-spark_3_2-12/Samples/Scala/filteringDuringMigration.scala

```diff
@@ -0,0 +1,159 @@
+// Databricks notebook source
+val cosmosEndpoint = "<cosmos-endpoint>"
+val cosmosMasterKey = "<cosmos-master-key>"
+val cosmosDatabaseName = "ContosoHospital"
+val cosmosContainerName = "Patient"
+
+
+// Patient Document -- partition key: patientId
+// {
+//   "id": "9c9a1156-e936-40f3-a442-e9528b55a2fb",
+//   "patientId": "423ab2cf-dd1c-4404-8524-86cee045c179",
+//   "patientName": "John Doe",
+//   "doctorId" : "629f49da-9cfc-45a4-8e1c-d4f8b7ab1f4e",
+//   "doctorName": "Sung Ondricka"
+// }
+
+
+//-----filtering examples-with-schema-inference-disabled---------------------
+
+val TargetContainerName = "CopyContainer"
+val checkpointLocation = "/tmp/streaming_checkpoint"
+
+val changeFeedCfg = Map("spark.cosmos.accountEndpoint" -> cosmosEndpoint,
+  "spark.cosmos.accountKey" -> cosmosMasterKey,
+  "spark.cosmos.database" -> cosmosDatabaseName,
+  "spark.cosmos.container" -> cosmosContainerName,
+  "spark.cosmos.read.inferSchema.enabled" -> "false",   
+  "spark.cosmos.changeFeed.startFrom" -> "Beginning",
+  "spark.cosmos.changeFeed.mode" -> "Incremental",
+  "spark.cosmos.changeFeed.itemCountPerTriggerHint" -> "10000"
+)
+
+val writeCfg = Map("spark.cosmos.accountEndpoint" -> cosmosEndpoint,
+  "spark.cosmos.accountKey" -> cosmosMasterKey,
+  "spark.cosmos.database" -> cosmosDatabaseName,
+  "spark.cosmos.container" -> TargetContainerName,
+  "checkpointLocation" -> checkpointLocation
+)
+
+// ----- EXAMPLE 1 -----
+
+
+val changeFeedDF = spark.readStream.format("cosmos.oltp.changeFeed")
+      .options(changeFeedCfg)  
+      .load
+
+println("Streaming DataFrame : " + changeFeedDF.isStreaming)
+
+
+import org.apache.spark.sql.functions.{split}
+// Here is an example of splitting the id column by "-" and creating a new column with only the second item
+// e.g. Id = XXXXX-SubIdWeAreInterestedIn-XXXXX
+val splitByDash = changeFeedDF.select($"*", split($"id","-").alias("idSplit"))
+val withSplitId = splitByDash.withColumn("newId", $"idSplit".getItem(1))
+
+// Filter by newId to get only the columns we want to migrate, then drop helper columns
+val filteredByNewId = withSplitId.filter($"newId" === 7926).drop("idSplit").drop("newId")
+
+// preserve system properties like _ts, _etag by renaming the original column
+val df_withAuditFields = filteredByNewId.withColumnRenamed("_rawbody", "_origin_rawBody")
+
+// write streaming dataframe to the target container
+df_withAuditFields
+  .writeStream
+  .format("cosmos.oltp")
+  .options(writeCfg)
+  .start()
+  .awaitTermination()
+
+
+
+// ----- EXAMPLE 2 -----
```

Comment:
This example is terribly inefficient - IMO it hurts more to have it

Reply:
removed